src/main/java/org/terasology/minecarts/components/CartJointComponent.java

```diff
@@ -3,19 +3,33 @@
 
 package org.terasology.minecarts.components;
 
-import org.terasology.engine.entitySystem.Component;
 import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.gestalt.entitysystem.component.Component;
 import org.terasology.reflection.MappedContainer;
 
-public class CartJointComponent implements Component {
+public class CartJointComponent implements Component<CartJointComponent> {
     public CartJointSocket front;
     public CartJointSocket back;
 
+    @Override
+    public void copy(CartJointComponent other) {
+        this.front = other.front.copy();
+        this.back = other.back.copy();
+    }
+
     @MappedContainer
     public static class CartJointSocket {
         public EntityRef entity;
         public boolean isOwning = false;
         public float range = .5f;
+
+        CartJointSocket copy() {
+            CartJointSocket newSocket = new CartJointSocket();
+            newSocket.entity = this.entity;
+            newSocket.isOwning = this.isOwning;
+            newSocket.range = this.range;
+            return newSocket;
+        }
```

Comment:
See [discussion on Discord](https://discord.com/channels/270264625419911192/577194115725262912/869675280980115486). 

Main question here is whether we want to use `a = b.copy()` or the same pattern as _gestalt_ introduces for the `copy(other)` method.